### DISCUSSION TITLE
Large models - performance

### DISCUSSION THREAD

**Discussion by barvirm** [bevyengine/bevy#9007] [Q&A] [answered]
Hello, I'm trying to render a massive model (GBL 900 MB, 14.5M triangles) with Bevy and I'm getting output with 60 FPS in a release.
When I tried to write a simple renderer in C++ with OpenGL, I was able to get around 300 fps without any problems. Is there any specialty on how to handle large models in Bevy to get at least close to 200+ fps?

C++ renderer, single draw call with Phong.

For Bevy I used: load_gltf.rs example, without DirectionalLightBundle, EnvironmentMapLight and animation.

**Comment by nicopap** [comment]
It's probably the fragment shader. I expect a lot of overdraw with tinny meshes. Overdraw wouldn't be a problem with a trivial fragment shader that doesn't do a lot, but bevy's pbr is a bit heavy.

Try disabling MSAA and see where this leads you: <https://docs.rs/bevy/latest/bevy/prelude/enum.Msaa.html>. It will drastically reduce the amount of overdraw.

**Edit**: Also try the `scene_viewer` for loading gltfs, just pass the asset path as argument `cargo run --example scene_viewer path/to/file.glb`

**Reply by barvirm** [reply]
I'm still getting 60 FPS, MSAA makes no difference.

**Reply by rparrett** [reply, marked as answer]
Exactly 60? That sounds like you're just being limited by vsync. You can turn off vsync by modifying `DefaultPlugins` like so:

```rust
.add_plugins(DefaultPlugins.set(WindowPlugin {
    primary_window: Some(Window {
        present_mode: PresentMode::AutoNoVsync,
        ..default()
    }),
    ..default()
}))
```

**Comment by barvirm** [comment]
Thank you all!!
